app/src/main/java/org/mozilla/fenix/home/recenttabs/controller/RecentTabController.kt

```diff
@@ -45,9 +45,17 @@ class DefaultRecentTabsController(
     }
 
     override fun handleRecentTabShowAllClicked() {
+        dismissSearchDialogIfDisplayed()
+
         navController.nav(
             R.id.homeFragment,
             HomeFragmentDirections.actionGlobalTabsTrayFragment()
         )
     }
+
+    private fun dismissSearchDialogIfDisplayed() {
+        if (navController.currentDestination?.id == R.id.searchDialogFragment) {
+            navController.navigateUp()
```

Comment:
I think we might want to use `view?.hideKeyboard()` here instead of `.navigateUp()`

Reply:
I also think we can just put these lines in `handleRecentTabShowAllClicked()` since it isn't being used elsewhere 😄

Reply:
@VitalyVPinchuk you can probably call `view?.hideKeyboard()` before we call the interactor. https://searchfox.org/mozilla-mobile/source/fenix/app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabsHeaderViewHolder.kt#25. This will avoid having us to pass the `view` into the controller.

Reply:
Thanks!
I tried to call `hideKeyboard()` in the exact same place, but nothing happened despite applying the method to `view`, to `rootView`, to `parent` and what not :(

Reply:
I've looked into `handleSelectTopSite()` and `onScrollStateChanged()` of `RecyclerView.OnScrollListener()` implementation and found that `navigateUp()` was used.
I just thought that `dismissSearchDialogIfDisplayed()` was much more self explaining than those lines of code. There's also `SessionControllController.kt`, which uses the same code, and tight coupling isn't a good thing so I just copied it 😄